### ISSUE TITLE
a11y feature request: DataTable select aria label

### ISSUE THREAD

**Issue by dgreene1** [grommet/grommet#6204]
If you look at [Success Criterion 3.3.2: Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html) you'll see that it says:

> In the case of radio buttons, **checkboxes,** comboboxes, or similar controls that provide users with options, each option must have an appropriate label so that users know what they are actually selecting

The emphasis above is mine because I'm expressing that the checkboxes on [data table](https://storybook.grommet.io/?path=/story/visualizations-datatable-grouped-and-onselect--grouped-on-select-data-table) doesn't read to the user what they are selecting. So the Data Table does not meet Success Criterion 3.3.2.

### Expected Behavior

When navigating the [group/on-select example](https://storybook.grommet.io/?path=/story/visualizations-datatable-grouped-and-onselect--grouped-on-select-data-table) with the NVDA screen reader, I would be able to hear what the checkbox for each selection is referring to.

### Actual Behavior

When tabbing to the checkbox, NVDA reads "blank." When navigating to the checkbox via the NVDA shortcut ctrl+alt+arrow, it reads "row 2 clickable." Neither of these are helpful for a visually impaired user since they don't know what it is in that row that they would be clicking.

### URL, screen shot, or Codepen exhibiting the issue

n.a. since it's non-visual

### Steps to Reproduce

1. open up NVDA
2. navigate to https://storybook.grommet.io/?path=/story/visualizations-datatable-grouped-and-onselect--grouped-on-select-data-table
3. tab until you hear "blank" for the second time
5. press ctrl+alt+up
6. press ctrl+alt+down and you'll hear "row 2 clickable"

### Your Environment

- Grommet version: whatever was in the storybook website on 6/27/2021
- Browser Name and version: 101.0.1 (64-bit)
- Operating System and version (desktop or mobile): Windows Version 10.0.19042

**Comment by jcfilben** [comment]
@dgreene1 Glad to hear about your team's commitment to accessibility and open source! I went ahead and filed a pull request for this issue since it was a pretty quick fix. We are a small team of people that maintain Grommet so we do rely a lot on community contributions so it is great to hear that your team is willing to contribute! In the future feel free to make contributions like this even if you don't have voice over. During the review process our team will test with voice over to make sure there aren't any issues. I also just want to mention that accessibility is a priority for us and we are striving to improve accessibility on Grommet as much as we can. Filing issue like this one is super helpful for our team!

**Comment by dgreene1** [comment]
Wow thank you so much @jcfilben! :) :) :)
